setup.py

```diff
@@ -213,11 +213,30 @@ def write_script(self, script_name, contents, mode="t", *ignored):
             contents = contents.replace(shebang, quoted_shebang)
         super().write_script(script_name, contents, mode, *ignored)
 
+
+class install_scripts_rewrite(install_scripts):
+    """Rewrite python3 to the current python executable"""
+    def run(self):
+        super().run()
+        if os.name == 'posix' and not self.dry_run:
+            for file in self.get_outputs():
+                if file.endswith('xon.sh'):
+                    bs_cmd = self.get_finalized_command('build_scripts')
+                    exec_param = getattr(bs_cmd, 'executable', None)
+
+                    with open(file, 'r') as f:
+                        content = f.read()
+
+                    processed = content.replace(' python3 ', ' {} '.format(exec_param))
```

Comment:
I think the format should be `' "{}" '.fomat(exec_param)`, to allow for bad actor cases such as when the user has a space in the abspath to the Python executable.